swe_roe/ex2b.c

```diff
@@ -1243,11 +1274,14 @@ static PetscErrorCode CreateDM(RDyApp app) {
   PetscCall(DMSetUseNatural(app->dm, PETSC_TRUE));
 
   // Distrubte the DM
-  DM dmDist;
-  PetscCall(DMPlexDistribute(app->dm, 1, NULL, &dmDist));
+  DM      dmDist;
+  PetscSF sfMigration;
```

Comment:
@bishtgautam Can you please add some comments why `sfMigration` is needed here?

Reply:
Unfortunately, I can't comment on why the `sfMigration` is needed because I have absolutely no idea. It is all about the DMPlex's magic that happens behind the scheme and I'm okay with not understanding it as long @knepley understands it (🤣).

Reply:
@donghuix The `sfMigration` is the map from the old (natural) mesh distribution to the new (petsc) mesh distribution. We need to keep it around in order to map other fields back to the natural mesh, like the auxiliary fields.

Reply:
@knepley Thanks for the explanation.